### DISCUSSION TITLE
Drop support for PHP 5.6?

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [Codeception/Codeception#6008] [Ideas]
Wouldn't you say it's about time to drop support for PHP 5.6 and go for 7.2+? https://github.com/Codeception/Codeception/blob/4.1/composer.json#L18
It has reached end-of-line almost 2 years ago: https://www.php.net/supported-versions.php

Just noticed this when looking at https://github.com/Codeception/module-rest/issues/24 ...

**Comment by Naktibalda** [comment]
Fine with me if it happens in Codeception 5.

**Comment by ThomasLandauer** [comment]
There is no 5.0 branch yet - could you create one?

**Comment by Naktibalda** [comment]
Dropping support for PHP 5.6 is not a big enough feature to mandate release of a new major version.

**Comment by ThomasLandauer** [comment]
I didn't mean to *release* it. What's your plan to collect all changes targeted v5?

**Comment by SamMousa** [comment]
In my opinion, it makes more sense to drop support for all but the most recent version when we decide to do Codeception 5.

Personally I'd wait for a PHP8 stable release, then start a Codeception 5 branch that targets _just_ PHP8 +.

**Comment by Naktibalda** [comment]
I would wait for PHPUnit 10 and support only it.

**Comment by Naktibalda** [comment]
As I wrote, it will happen in Codeception 5: https://github.com/Codeception/Codeception/discussions/6064

**Comment by ThomasLandauer** [comment]
OK, great!
Is there a way to close this discussion?

**Comment by TavoNiievez** [comment]
Hey,
There is little more than 3 months of support for PHP 7.3, surely it will be much less when Codeception 5 is released. Do we still want that version to be compatible?

**Comment by c33s** [comment]
i work quite a lot with legacy projects. my personal opinion is to keep tools like testing frameworks, composer, rector,... as compatible as possible with old version. this is the only way to migrate a legacy 5.6 project to a higher php version.
